react/Carousel.tsx

```diff
@@ -290,71 +90,255 @@ export default class Carousel extends Component<Props, State> {
+      autoplay={autoplay && autoplayInterval > 0}
+      autoplayInterval={autoplayInterval * 1000}
+      pauseOnHover
+      onNextSlide={handleNextSlide}
+      className={styles.container}
+    >
+      <Slider
+        loop
+        classes={{
+          root: handles.sliderRoot,
+          sliderFrame: handles.sliderFrame,
+        }}
+        perPage={PER_PAGE}
+        arrowRender={showArrows && ArrowRender}
+        currentSlide={currentSlide}
+        onChangeSlide={setCurrentSlide}
+        arrowsContainerComponent={showArrows && ArrowContainerRender}
+        duration={500}
       >
-        <Slider
+        {banners.map((banner, i) => (
+          <Slide
+            className={handles.slide}
+            key={i}
+            style={{ maxHeight: height }}
+            sliderTransitionDuration={500}
+          >
+            <Banner height={height} {...banner} />
+          </Slide>
+        ))}
+      </Slider>
+      {showDots && (
+        <Dots
           loop
+          perPage={PER_PAGE}
+          currentSlide={currentSlide}
+          totalSlides={banners.length}
+          onChangeSlide={setCurrentSlide}
           classes={{
-            root: styles.sliderRoot,
-            sliderFrame: styles.sliderFrame,
+            activeDot: classnames(handles.activeDot, 'bg-emphasis'),
+            dot: classnames(styles.dot, 'mh2 mv0 pointer br-100'),
+            notActiveDot: classnames(handles.notActiveDot, 'bg-muted-3'),
+            root: classnames(handles.containerDots, 'bottom-0 pb4'),
           }}
-          perPage={this.perPage}
-          arrowRender={showArrows && this.ArrowRender}
-          currentSlide={currentSlide}
-          onChangeSlide={this.handleChangeSlide}
-          arrowsContainerComponent={showArrows && this.ArrowContainerRender}
-          duration={500}
-        >
-          {banners.map((banner, i) => (
-            <Slide
-              className={styles.slide}
-              key={i}
-              style={{ maxHeight: height }}
-              sliderTransitionDuration={500}
-            >
-              <Banner height={height} {...banner} />
-            </Slide>
-          ))}
-        </Slider>
-        {showDots && (
-          <Dots
-            loop
-            perPage={this.perPage}
-            currentSlide={currentSlide}
-            totalSlides={banners.length}
-            onChangeSlide={this.handleChangeSlide}
-            classes={{
-              activeDot: classnames(styles.activeDot, 'bg-emphasis'),
-              dot: classnames(styles.dot, 'mh2 mv0 pointer br-100'),
-              notActiveDot: classnames(styles.notActiveDot, 'bg-muted-3'),
-              root: classnames(styles.containerDots, 'bottom-0 pb4'),
-            }}
-          />
-        )}
-      </SliderContainer>
-    )
+        />
+      )}
+    </SliderContainer>
+  )
+}
+
+Carousel.defaultProps = {
+  autoplay: true,
+  autoplaySpeed: 5,
+  banners: [],
+  height: 420,
+  showArrows: true,
+  showDots: true,
+}
```

Comment:
Kind of a nitpick, but why use `defaultProps` and not just set the default values when you're destructuring the `props` object? I think it makes it more readable since you don't need to scroll all the way here to see the default values of each prop